pkg/prometheus/promcfg_test.go

```diff
@@ -8879,6 +8879,7 @@ func TestScrapeConfigSpecConfig(t *testing.T) {
   external_labels:
     prometheus: default/test
     prometheus_replica: $(POD_NAME)
+  metric_relabel_configs: []
```

Comment:
We shouldn't add an empty slice if `spec.MetricRelabelConfigs` wasn't set, see my previous comment. This is true for all the tests were you added this.